### PR TITLE
feat(FN-3667): pending corrections list on portal

### PR DESCRIPTION
**Pull request by RLavender98**
## Introduction :pencil2:
When a PDC user has requested a correction we need to give the bank users a way to view the request and action it. This will happen by when they go to the report upload page if there are any correction requests that need a response displaying them in a table instead of displaying the report upload. 

## Resolution :heavy_check_mark:
- Add a new central-api endpoint to fetch all pending corrections from the oldest report with pending corrections for a bank.
- Update the get upload controller to check if the feature flag is enabled and if so check for pending corrections and if there are any render the pending corrections template instead of the report upload template.
- New pending corrections template for the new view of the page.

## Miscellaneous :heavy_plus_sign:
- Rewrite some existing controllers and tests written in javascript in typescript
- e2e-test set up for record correction tests in portal
- Rename `validateTfmFeeRecordCorrectionsFeatureFlagEnabled` validation middleware to `validateFeeRecordCorrectionsFeatureFlagEnabled`

## Out of scope of PR
I have not include the reported fees and fees paid to ukef in the reported period columns because there is an ongoing discussion about what data should be displayed in these columns. A follow up PR will be raised once the requirement has been finalised.

## Screenshot
With report due for upload:
Screenshot 2024-12-18 at 15 08 00

Without a report due for upload:
Screenshot 2024-12-18 at 15 05 29

### CODE REVIEW

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+    return mapFeeRecordToPendingCorrectionsArray(feeRecord);
+  });
+};
+
+/**
+ * Gets the next due report period for a bank with pending corrections,
+ *  where the next due report period is defined as:
+ * - The report period for the oldest not received report
+ * - The next report period in the bank's schedule if there are no not received reports
```
[the remark this answers is not in the document]
Comment (RLavender98) [reply dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers.ts:45]: Have gone with explicitly using the report status

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/index.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    if (!bank.isVisibleInTfmUtilisationReports) {
+      throw new NotFoundError(`Bank with id '${bankId}' is not opted into utilisation reporting via DTFS`);
+    }
+
+    const oldestReportWithPendingCorrections = await UtilisationReportRepo.findOldestReportWithPendingCorrectionsByBankId(bankId);
+
+    if (!oldestReportWithPendingCorrections) {
+      // If there aren't any reports with pending corrections then we return null.
```
[the remark this answers is not in the document]
Comment (RLavender98) [reply dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/index.ts:62]: I think what I was trying to explain was that if there is no oldest report with pending corrections, then there are no reports at all with pending corrections. I have reworded a little, let me know if you still think it is unnecessary.

File: portal/component-tests/utilisation-report-service/record-corrections/pending-corrections.component-test.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+      wrapper.expectElement('[data-cy="pending-corrections-table"]').toExist();
+    });
+
+    it('should render the corrections table headers', () => {
+      // Act
+      const wrapper = render(viewModel);
+
+      // Assert
+      wrapper.expectElement('th').toHaveCount(3);
+      wrapper.expectElement('th:contains("Facility ID")').toExist();
```
Comment (sw-mattw) [inline portal/component-tests/utilisation-report-service/record-corrections/pending-corrections.component-test.js:176]: MATT: NOT SURE ON THIS

File: portal/templates/utilisation-report-service/record-corrections/pending-corrections.njk [hunk 4]
```diff
@@ -0,0 +1,82 @@
+{% extends "index.njk" %}
+{% from "govuk/components/table/macro.njk" import govukTable %}
+
+{% block pageTitle %}Report GEF utilisation and fees - Record correction{% endblock %}
```
[the remark this answers is not in the document]
Comment (sw-mattw) [reply portal/templates/utilisation-report-service/record-corrections/pending-corrections.njk:4]: Thanks

File: portal/component-tests/utilisation-report-service/record-corrections/pending-corrections.component-test.js [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+      wrapper.expectElement('th:contains("Facility ID")').toExist();
+      wrapper.expectElement('th:contains("Facility ID")').toExist();
```
Comment (sw-mattw) [inline portal/component-tests/utilisation-report-service/record-corrections/pending-corrections.component-test.js:178]: Second and third should be `Exporter` and `Error summary`

**Review by sw-mattw — CHANGES_REQUESTED** [review]

**Review by sw-mattw — DISMISSED** [review]

File: dtfs-central-api/api-tests/v1/utilisation-reports/fee-record-corrections/get-utilisation-report-pending-corrections.api-test.ts [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+      .withBankId(bankId)
+      .withUploadedByUserId(portalUserId)
+      .withDateUploaded(new Date('2021-01-01'))
+      .build();
+    const feeRecord = FeeRecordEntityMockBuilder.forReport(report)
+      .withStatus(FEE_RECORD_STATUS.PENDING_CORRECTION)
+      .withFacilityId('1122334455')
+      .withExporter('Test Exporter')
+      .build();
+    const completedCorrection = FeeRecordCorrectionEntityMockBuilder.forFeeRecord(feeRecord)
```
Comment (Zainzzkk) [inline dtfs-central-api/api-tests/v1/utilisation-reports/fee-record-corrections/get-utilisation-report-pending-corrections.api-test.ts:120]: new lines please between these

File: dtfs-central-api/api-tests/v1/utilisation-reports/fee-record-corrections/get-utilisation-report-pending-corrections.api-test.ts [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+      expect(response.status).toEqual(HttpStatusCode.Ok);
+    });
+
+    it('should return the pending corrections details for the oldest report with pending corrections', async () => {
+      // Act
+      const response = await testApi.get(replaceUrlParameterPlaceholders(BASE_URL, { bankId }));
+
+      // Assert
+      expect(response.body).toEqual({
+        reportPeriod: oldestReportWithPendingCorrections.reportPeriod,
```
Comment (Zainzzkk) [inline dtfs-central-api/api-tests/v1/utilisation-reports/fee-record-corrections/get-utilisation-report-pending-corrections.api-test.ts:269]: can this be an expected object please

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers.test.ts [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+      ];
+
+      // Act
+      const result = mapFeeRecordsToPendingCorrections(feeRecords);
+
+      // Assert
+      expect(result).toEqual([]);
+    });
+
+    it(`should return an array of all of the pending corrections of the fee records`, () => {
```
Comment (Zainzzkk) [inline dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers.test.ts:151]: can more be added to this it (like the ones above) for why an array is being returned

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers.test.ts
Comment (Zainzzkk) [inline dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers.test.ts:1]: this test file is very long and hard to follow.  Can it be split into one for each helper?

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers.ts [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+ * @param bank - The bank
+ * @returns The next due report period
+ */
+export const getNextDueReportPeriod = async (bank: Bank): Promise<ReportPeriod> => {
+  const dueReports = await UtilisationReportRepo.findDueReportsByBankId(bank.id);
+
+  if (dueReports.length === 0) {
+    return getNextReportPeriodForBankSchedule(bank.utilisationReportPeriodSchedule);
+  }
+  return dueReports[0].reportPeriod;
```
Comment (Zainzzkk) [inline dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers.ts:56]: new lines please in this file

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers.ts [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+};
+
+/**
+ * Maps a report with pending corrections to the pending corrections response body
+ * @param report - The report
+ * @param bank - The bank
+ * @returns The pending corrections response body
+ */
+export const mapReportToPendingCorrectionsResponseBody = async (report: UtilisationReportEntity, bank: Bank): Promise<PendingCorrectionsResponseBody> => {
+  if (report.uploadedByUserId === null || report.dateUploaded === null) {
```
Comment (Zainzzkk) [inline dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers.ts:66]: could this be `!report.uploadedByUserId  || !report.dateUploaded`

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/index.test.ts [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+        // Act
+        await getUtilisationReportPendingCorrectionsByBankId(req, res);
+
+        // Assert
+        expect(res.statusCode).toEqual(HttpStatusCode.NotFound);
+        expect(res._getData()).toEqual(`Failed to get pending corrections: Failed to find bank with id '${bankId}'`);
+      });
+    });
+
+    describe('when the bank is not opted into utilisation reporting via DTFS', () => {
```
Comment (Zainzzkk) [inline dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/index.test.ts:61]: should this be has not?

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/index.ts [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+    const { bankId } = req.params;
+
+    const bank = await getBankById(bankId);
+
+    if (!bank) {
+      throw new NotFoundError(`Failed to find bank with id '${bankId}'`);
+    }
+
+    if (!bank.isVisibleInTfmUtilisationReports) {
+      throw new NotFoundError(`Bank with id '${bankId}' is not opted into utilisation reporting via DTFS`);
```
Comment (Zainzzkk) [inline dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/index.ts:56]: has not?

File: e2e-tests/portal/cypress/e2e/journeys/payment-report-officer/record-corrections/feature-flag-disabled/pending-corrections.spec.js [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+  PENDING_RECONCILIATION,
+  FeeRecordCorrectionEntityMockBuilder,
+  FeeRecordEntityMockBuilder,
+  FEE_RECORD_STATUS,
+} = require('@ukef/dtfs2-common');
+const { NODE_TASKS, BANK1_PAYMENT_REPORT_OFFICER1 } = require('../../../../../../../e2e-fixtures');
+const relativeURL = require('../../../../relativeURL');
+const { pendingCorrections } = require('../../../../pages');
+
+context('Pending corrections - Fee record correction feature flag enabled', () => {
```
Comment (Zainzzkk) [inline e2e-tests/portal/cypress/e2e/journeys/payment-report-officer/record-corrections/feature-flag-disabled/pending-corrections.spec.js:12]: this is in feature flag enabled folder

File: e2e-tests/portal/cypress/e2e/journeys/payment-report-officer/record-corrections/feature-flag-disabled/pending-corrections.spec.js [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+        const bankId = bank.id;
+
+        const report = UtilisationReportEntityMockBuilder.forStatus(PENDING_RECONCILIATION).withUploadedByUserId(_id.toString()).withBankId(bankId).build();
+        const feeRecord = FeeRecordEntityMockBuilder.forReport(report).withStatus(FEE_RECORD_STATUS.PENDING_CORRECTION).build();
+        const pendingCorrection = FeeRecordCorrectionEntityMockBuilder.forFeeRecord(feeRecord).withId(1).withIsCompleted(false).build();
+
+        cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+        cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [feeRecord]);
+        cy.task(NODE_TASKS.INSERT_FEE_RECORD_CORRECTIONS_INTO_DB, [pendingCorrection]);
+      });
```
Comment (Zainzzkk) [inline e2e-tests/portal/cypress/e2e/journeys/payment-report-officer/record-corrections/feature-flag-disabled/pending-corrections.spec.js:34]: can this be in the before so the it statement is much shorter

File: e2e-tests/portal/cypress/e2e/journeys/payment-report-officer/record-corrections/feature-flag-enabled/pending-corrections.spec.js [hunk 15]
```diff
@@ -0,0 +1,196 @@
...
+          const { _id, bank } = user;
+          const bankId = bank.id;
+
+          const report = UtilisationReportEntityMockBuilder.forStatus(PENDING_RECONCILIATION)
+            .withId(1)
+            .withUploadedByUserId(_id.toString())
+            .withBankId(bankId)
+            .withReportPeriod(pendingCorrectionReportPeriod)
+            .build();
+          const feeRecord = FeeRecordEntityMockBuilder.forReport(report)
```
Comment (Zainzzkk) [inline e2e-tests/portal/cypress/e2e/journeys/payment-report-officer/record-corrections/feature-flag-enabled/pending-corrections.spec.js:49]: new lines so easier to read please

File: e2e-tests/portal/cypress/e2e/journeys/payment-report-officer/record-corrections/feature-flag-enabled/pending-corrections.spec.js [hunk 16]
```diff
@@ -0,0 +1,196 @@
...
+          pendingCorrections.noReportDueText().should('not.exist');
+        });
+      });
+
+      context('and when there are no reports due for upload', () => {
+        const formattedNextDueReportPeriodAlias = 'formattedNextDueReportPeriod';
+        const formattedUploadFromDateAlias = 'formattedUploadFromDate';
+
+        beforeEach(() => {
+          cy.task('getUserFromDbByEmail', BANK1_PAYMENT_REPORT_OFFICER1.email).then((user) => {
```
Comment (Zainzzkk) [inline e2e-tests/portal/cypress/e2e/journeys/payment-report-officer/record-corrections/feature-flag-enabled/pending-corrections.spec.js:136]: could this be in a before and the login in a beforeEach so we are not doing this call for every it statement?

File: portal-api/src/v1/api-response-types/utilisation-report-pending-corrections-response-body.ts [hunk 17]
```diff
@@ -0,0 +1,18 @@
+import { IsoDateTimeStamp, ReportPeriod } from '@ukef/dtfs2-common';
+
+type PendingCorrection = {
+  correctionId: number;
+  facilityId: string;
+  exporter: string;
+  additionalInfo: string;
+};
```
Comment (Zainzzkk) [inline portal-api/src/v1/api-response-types/utilisation-report-pending-corrections-response-body.ts:8]: libs/common? also being used in `dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/index.ts`

File: portal-api/src/v1/api-response-types/utilisation-report-pending-corrections-response-body.ts [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+
+export type UtilisationReportPendingCorrectionsResponseBody =
```
Comment (Zainzzkk) [inline portal-api/src/v1/api-response-types/utilisation-report-pending-corrections-response-body.ts:10]: could this also be in libs/common?

File: portal-api/src/v1/controllers/utilisation-report-service/fee-record-corrections/pending-corrections.ts [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+ * @param req - The request object
+ * @param res - The response object
+ */
+export const getUtilisationReportPendingCorrectionsByBankId = async (req: GetUtilisationReportPendingCorrectionsRequest, res: Response) => {
+  try {
+    const { bankId } = req.params;
+
+    const pendingCorrections = await api.getUtilisationReportPendingCorrectionsByBankId(bankId);
+
+    return res.status(200).send(pendingCorrections);
```
Comment (Zainzzkk) [inline portal-api/src/v1/controllers/utilisation-report-service/fee-record-corrections/pending-corrections.ts:25]: HttpStatus.ok instead?

File: portal/component-tests/utilisation-report-service/record-corrections/pending-corrections.component-test.js [hunk 20]
```diff
@@ -0,0 +1,210 @@
...
+    };
+
+    // Act
+    const wrapper = render(viewModel);
+
+    // Assert
+    wrapper
+      .expectText('[data-cy="corrections-info-text"]')
+      .toRead(
+        `The following errors in the ${viewModel.formattedReportPeriod} report uploaded by ${viewModel.uploadedByFullName} on ${viewModel.formattedDateUploaded} have been identified by UKEF:`,
```
Comment (Zainzzkk) [inline portal/component-tests/utilisation-report-service/record-corrections/pending-corrections.component-test.js:32]: could this be a const since it's a long string being constructed?

File: portal/server/controllers/utilisation-report-service/utilisation-report-upload/pending-corrections-helper.test.ts [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+    });
+  });
+
+  describe('isNonEmptyPendingCorrectionsResponseBody', () => {
+    it('should return true if the response body is not an empty object', () => {
+      const responseBody = aNonEmptyPendingCorrectionsResponseBody();
+
+      const result = isNonEmptyPendingCorrectionsResponseBody(responseBody);
+
+      expect(result).toBe(true);
```
Comment (Zainzzkk) [inline portal/server/controllers/utilisation-report-service/utilisation-report-upload/pending-corrections-helper.test.ts:105]: toEqual?

File: portal/test-helpers/test-data/portal-session-bank.ts
Comment (Zainzzkk) [inline portal/test-helpers/test-data/portal-session-bank.ts:1]: this already exists in libs/common/src/test-helpers/bank.ts

File: portal/test-helpers/test-data/report-period.ts
Comment (Zainzzkk) [inline portal/test-helpers/test-data/report-period.ts:1]: could this also go in libs/common.  could be reused

File: portal/test-helpers/test-data/utilisation-reporting-schedule.ts
Comment (Zainzzkk) [inline portal/test-helpers/test-data/utilisation-reporting-schedule.ts:1]: this as well

File: e2e-tests/portal/cypress/e2e/pages/utilisation-report-service/record-corrections/pendingCorrections.js [hunk 22]
```diff
@@ -0,0 +1,16 @@
+const page = {
+  table: () => cy.get('[data-cy="pending-corrections-table"]'),
+  rows: () => cy.get(`[data-cy="pending-corrections-table"] tr`),
+  row: (index) => ({
+    facilityId: () => cy.get(`[data-cy="pending-corrections-table"] tr:nth-child(${index}) [data-cy="pending-corrections-row--facility-id"]`),
```
Comment (Zainzzkk) [inline e2e-tests/portal/cypress/e2e/pages/utilisation-report-service/record-corrections/pendingCorrections.js:5]: could each row not use a correctionId so we don't need to find by row index?

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/index.test.ts (hunk 11 above)
Comment (RLavender98) [reply dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/index.test.ts:61]: I guess semantically the bank doesn't opt themselves in which I would expect to be the case with the wording 'the bank has'

File: e2e-tests/portal/cypress/e2e/journeys/payment-report-officer/record-corrections/feature-flag-enabled/pending-corrections.spec.js (hunk 16 above)
Comment (RLavender98) [reply e2e-tests/portal/cypress/e2e/journeys/payment-report-officer/record-corrections/feature-flag-enabled/pending-corrections.spec.js:136]: I don't think so because I can't see a way to clean up that doesn't use an afterEach deleting everything without deleting specific entities which is not something we should do because that is harder to maintain and less reliable

File: portal-api/src/v1/api-response-types/utilisation-report-pending-corrections-response-body.ts (hunk 18 above)
Comment (RLavender98) [reply portal-api/src/v1/api-response-types/utilisation-report-pending-corrections-response-body.ts:10]: We haven't been doing this for types which differ in central api because the response body is quite tied to the service and we don't want people getting confused about which services they belong in, e.g. this type is different in central-api because in central-api the date uploaded is a `Date` but in the apis consuming it it will have been stringified into an `IsoDateTimeStamp` so the types are different in different services.

File: portal/test-helpers/test-data/report-period.ts
Comment (RLavender98) [reply portal/test-helpers/test-data/report-period.ts:1]: I'm not sure we should be putting so many test only helpers into libs common because they will get packaged up with the actual main code when the project is built and increase the size of the packaged library no? Or is that not a concern?

File: e2e-tests/portal/cypress/e2e/pages/utilisation-report-service/record-corrections/pendingCorrections.js (hunk 22 above)
Comment (RLavender98) [reply e2e-tests/portal/cypress/e2e/pages/utilisation-report-service/record-corrections/pendingCorrections.js:5]: I guess we could, but I thought this was pretty nice and will be useful when we are doing e2e tests for the full complete journey where we won't know what the id is.

File: portal/test-helpers/test-data/report-period.ts
Comment (Zainzzkk) [reply portal/test-helpers/test-data/report-period.ts:1]: I am thinking that this may be reused in another microservice such as central-api? would be better than replication.  Can keep it here for now

File: e2e-tests/portal/cypress/e2e/pages/utilisation-report-service/record-corrections/pendingCorrections.js (hunk 22 above)
Comment (Zainzzkk) [reply e2e-tests/portal/cypress/e2e/pages/utilisation-report-service/record-corrections/pendingCorrections.js:5]: lets leave it for now.  I have something in my PR to find the record correction id so we can refactor later once my PR is in

**Review by Zainzzkk — DISMISSED** [review]

**Review by Zainzzkk — APPROVED** [review]

File: portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.ts [hunk 23]
```diff
@@ -1,15 +1,23 @@
...
+import { asLoggedInUserSession, LoggedInUserSession } from '../../../helpers/express-session';
+import { isNonEmptyPendingCorrectionsResponseBody, mapToPendingCorrectionsViewModel } from './pending-corrections-helper';
+
+/**
+ * Sets the utilisation report session data for the report upload journey
+ * @param req - The request object
+ * @param nextDueReportPeriod - The next due report period
+ */
+const setSessionUtilisationReport = (req: Request, nextDueReportPeriod: ReportPeriod & { formattedReportPeriod: string }) => {
+  (req.session as LoggedInUserSession).utilisationReport = {
```
Comment (Zainzzkk) [inline portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.ts:20]: from sonarCloud - This assertion is unnecessary since it does not change the type of the expression. (as LoggedInUserSession not needed)

File: portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.ts [hunk 24]
```diff
@@ -174,70 +220,94 @@ const postUtilisationReportUpload = async (req, res) => {
...
     const mappedReportData = removeCellAddressesFromArray(reportData);
 
     const response = await api.uploadUtilisationReportData(user, reportPeriod, mappedReportData, fileBuffer, formattedReportPeriod, userToken);
 
     if (response?.status === 200 || response?.status === 201) {
-      const { paymentOfficerEmails } = response.data;
-      req.session.utilisationReport = {
-        ...req.session.utilisationReport,
+      const { paymentOfficerEmails } = response.data as { paymentOfficerEmails: string[] };
+      (req.session as LoggedInUserSession).utilisationReport = {
```
Comment (Zainzzkk) [inline portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.ts:275]: same as above

File: portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.ts (hunk 23 above)
Comment (RLavender98) [reply portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.ts:20]: If I remove it I get the following:
Screenshot 2025-01-06 at 17 47 35

Comment (Zainzzkk) [reply portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.ts:20]: can ignore this one then

File: portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.ts (hunk 24 above)
Comment (RLavender98) [reply portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.ts:275]: Same as above

**Review by Zainzzkk — APPROVED** [review]

File: dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers/index.test.ts [hunk 25]
```diff
@@ -0,0 +1,114 @@
...
+              .build(),
+          ])
+          .build();
+      });
+
+      describe('and when finding the user errors', () => {
+        const mockError = new Error('Could not find user');
+
+        beforeEach(() => {
+          jest.mocked(getUserById).mockRejectedValue(mockError);
```
Comment (sw-mattw) [inline dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/helpers/index.test.ts:79]: [Failing tests](https://github.com/UK-Export-Finance/dtfs2/actions/runs/12637660031/job/35212362990?pr=4086) - Missing `jest.mock` for users-repo at the top of the file maybe?
